collector/library/pod.js

```diff
@@ -224,15 +210,57 @@ class Pod {
      * 
      */
     async submitDataToLogstash({ data, batch, }) {
-        const body = JSON.stringify({ message: data.toString() });
+        const dataAsString = data.toString();
+        let lines;
+
+        if (this.remainder.length > 0) {
+            lines = (this.remainder + dataAsString).split('\n');
+            this.remainder = '';
+        } else {
+            lines = dataAsString.split('\n');
+        }
 
-        await fetch('http://logstash:5000/', {
-            method: 'post',
-            body,
-            headers: { 'Content-Type': 'application/json' },
-        });
+        for (let i = 0; i < lines.length; i++) {
+            const line = lines[i];
+            let body;
+
+            if (this.isJSON(trim(line))) {
```

Comment:
instead of asking if it's json or not - insert the wrapping into `this.isJSON()` so that you don't need a condition:
```
const dataObject = this.jsonify(line);
dataObject._collector_batchId = batch.id;
dataObject._collector_source_url = this.targetUrl;
body = JSON.stringify(dataObject);
```

Reply:
Couldn't follow what you mean